### PR TITLE
Ipb 785/it2 v4

### PR DESCRIPTION
**Pull request by rgforsyth**
## What does this pull request do?

Updates the session feedback flow for both delivery sessions and saa sessions.

## What is the intent behind these changes?

To give more accurate data around if pop's attended a session.

### CODE REVIEW

File: server/routes/appointments/feedback/shared/sessionFeedback/noSessionFeedbackView.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+        classes: 'govuk-body govuk-!-margin-bottom-4',
+        isPageHeading: false,
+      },
+      value: this.inputsPresenter.fields.noSessionReasonLogistics.value,
+      errorMessage: ViewUtils.govukErrorMessage(this.inputsPresenter.fields.noSessionReasonLogistics.errorMessage),
+    }
+  }
+
+  // private get otherAdditionalInformationTextAreaArgs(): TextareaArgs {
+  //   return {
```
Comment (marcocolluramoj) [inline server/routes/appointments/feedback/shared/sessionFeedback/noSessionFeedbackView.ts:158]: Should we delete all the commented out code for noSessionReasonOther?

**Review by marcocolluramoj — APPROVED** [review]
LGTM

File: server/routes/appointments/feedback/actionPlanSessions/checkYourAnswers/actionPlanPostSessionFeedbackCheckAnswersPresenter.ts [hunk 2]
```diff
@@ -23,13 +23,30 @@ export default class ActionPlanPostSessionFeedbackCheckAnswersPresenter extends
     : `/service-provider/action-plan/${this.actionPlanId}/appointment/${this.actionPlanAppointment.sessionNumber}/post-session-feedback/submit`
 
   get backLinkHref(): string {
-    if (this.draftId) {
-      return this.actionPlanAppointment.appointmentFeedback.attendanceFeedback.attended === 'no'
-        ? `/service-provider/action-plan/${this.actionPlanId}/appointment/${this.actionPlanAppointment.sessionNumber}/post-session-feedback/edit/${this.draftId}/attendance`
-        : `/service-provider/action-plan/${this.actionPlanId}/appointment/${this.actionPlanAppointment.sessionNumber}/post-session-feedback/edit/${this.draftId}/behaviour`
+    if (this.actionPlanAppointment.appointmentFeedback.attendanceFeedback.didSessionHappen === true) {
```
Comment (kth13) [inline server/routes/appointments/feedback/actionPlanSessions/checkYourAnswers/actionPlanPostSessionFeedbackCheckAnswersPresenter.ts:26]: can this be `if (this.actionPlanAppointment.appointmentFeedback.attendanceFeedback.didSessionHappen) {`

File: server/routes/appointments/feedback/actionPlanSessions/checkYourAnswers/actionPlanPostSessionFeedbackCheckAnswersPresenter.ts [hunk 3]
```diff
@@ -23,13 +23,30 @@ export default class ActionPlanPostSessionFeedbackCheckAnswersPresenter extends
...
+      : `/service-provider/action-plan/${this.actionPlanId}/appointment/${this.actionPlanAppointment.sessionNumber}/post-session-feedback/no-session`
+  }
+
+  get attendanceFeedbackChangeLink(): string {
+    return this.draftId
+      ? `/service-provider/action-plan/${this.actionPlanId}/appointment/${this.actionPlanAppointment.sessionNumber}/post-session-feedback/edit/${this.draftId}/attendance`
+      : `/service-provider/action-plan/${this.actionPlanId}/appointment/${this.actionPlanAppointment.sessionNumber}/post-session-feedback/attendance`
+  }
+
+  get sessionFeedbackChangeLink(): string {
```
Comment (kth13) [inline server/routes/appointments/feedback/actionPlanSessions/checkYourAnswers/actionPlanPostSessionFeedbackCheckAnswersPresenter.ts:42]: Same comment as above

File: server/routes/appointments/feedback/shared/sessionFeedback/sessionFeedbackQuestionnaire.ts [hunk 4]
```diff
@@ -40,12 +40,55 @@ export default class SessionFeedbackQuestionnaire {
...
+    }
+  }
+
+  get noSessionReasonAdditionalInformationLogisticsQuestion(): { text: string } {
+    return {
+      text: `Add details about the issue.`,
+    }
+  }
+
+  // get noSessionReasonAdditionalInformationOtherQuestion(): { text: string } {
```
Comment (kth13) [inline server/routes/appointments/feedback/shared/sessionFeedback/sessionFeedbackQuestionnaire.ts:86]: can we remove the commented methods

**Review by kth13 — COMMENTED** [review]
Looks good with some minor questions and comments

**Review by kth13 — APPROVED** [review]
LGTM

**Pull request merged**